src/spec-configuration/configuration.ts

```diff
@@ -230,3 +239,48 @@ export async function getDockerComposeFilePaths(cliHost: FileHost, config: DevCo
 	}
 	return [];
 }
+
+export function buildExtendBehaviorTable(customBehaviors?: Partial<DevContainerExtendConfig>): DevContainerExtendConfig {
+	const defaultBehaviors: DevContainerExtendConfig = {
```

Comment:
What makes this difficult to use is that the user can't tell by a simple, easy to remember rule how things are going to be merged because it takes a case-by-case approach.

Reply:
I think their general idea (unless it has changed) is that arrays merged and objects/single values replaced by default.  However, this property can be overwritten (now sure how they landed on that) for specific use-cases.

Reply:
Sounds good. I see a few arrays listed as "REPLACE" and am wondering if that is intended. (Also curious about the `customBehaviors` override, but didn't see that being used.)

Reply:
We've thus far talked about the logic behaving something like:

1. Hahes/Objects merge
2. Arrays merge
3. Strings/Ints/etc replace

Would it be possible to refactor this be based around the types rather than the specific keys?  I could imagine this being fragile if, for example, a single-value key is extended to support an array-value.